src/microbe_stage/editor/MicrobeEditorAction.cs

```diff
@@ -27,29 +18,45 @@ public class MicrobeEditorAction : ReversibleAction
     [JsonProperty]
     private readonly MicrobeEditor editor;
 
-    public MicrobeEditorAction(MicrobeEditor editor, int cost,
+    private MicrobeEditorActionData data = null!;
```

Comment:
Isn't the null forgiving here unnecessary as the constructor initializes this field?
(maybe it should directly write to data if the write through Data is confusing the compiler).